spec/core/plugin_spec.rb

```diff
@@ -29,6 +29,22 @@ class LogStash::Filters::LadyGaga < LogStash::Filters::Base
     expect(LogStash::Plugin.lookup("filter", "lady_gaga")).to eq(LogStash::Filters::LadyGaga)
   end
 
+  describe "#inspect" do
+    class LogStash::Filters::MyTestFilter < LogStash::Filters::Base
```

Comment:
this is a class specific for this test, I don't see the pollution and having that in a separate file would make it really less clear. here we can straight away see that a test class is defined and the test is performed on it.